python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -1300,51 +1310,56 @@ def _legacy_process_configs(self, config_data, entity_type, project_id, all_acti
                 logger.error(
                     "A shotgun_get_actions call did not succeed: %s",
                     raw_actions_data
                 )
+
+                all_actions[config_name] = dict(
+                    actions=[],
+                    config=config_entity,
+                )
+
                 continue
 
             config_names.append(config_name)
 
             # The data returned by the tank command is a newline delimited string
             # that defines rows of ordered data delimited by $ characters.
-            for line in raw_actions_data["out"].split("\n"):
-                action = line.split("$")
+            try:
+                for line in raw_actions_data["out"].split("\n"):
+                    action = line.split("$")
 
-                if action[2] == "":
-                    deny_permissions = []
-                else:
-                    deny_permissions = action[2].split(",")
+                    if action[2] == "":
+                        deny_permissions = []
+                    else:
+                        deny_permissions = action[2].split(",")
 
-                multi_select = action[3] == "True"
+                    multi_select = action[3] == "True"
 
-                commands.append(
-                    dict(
-                        name=action[0],
-                        title=action[1],
-                        deny_permissions=deny_permissions,
-                        supports_multiple_selection=multi_select,
-                        app_name=None, # Not used here.
-                        group=None, # Not used here.
-                        group_default=None, # Not used here.
-                        engine_name=None, # Not used here.
+                    commands.append(
+                        dict(
+                            name=action[0],
+                            title=action[1],
+                            deny_permissions=deny_permissions,
+                            supports_multiple_selection=multi_select,
+                            app_name=None, # Not used here.
+                            group=None, # Not used here.
+                            group_default=None, # Not used here.
+                            engine_name=None, # Not used here.
+                        )
                     )
-                )
-
-            # We don't need or want the descriptor object to be sent to the client.
-            # Since we're done with this config for this invokation, we can just
-            # delete it from the entity dict.
-            del config_entity["descriptor"]
-
-            # And since we know this set of actions came from this legacy path, we
-            # can go ahead and include some extra data in the config dict that we
-            # can key off of when this action is called from the client.
-            config_entity[constants.LEGACY_CONFIG_ROOT] = config_path
+            except IndexError:
```

Comment:
i would prefer a broader catch here - we don't expect ANYTHING to go wrong in here, right? we are not calling any methods and we don't want anything to bubble up. Would also be nice to include what the error is and perhaps not say "legacy".